modules/reindex/src/main/java/org/elasticsearch/index/reindex/DeleteByQueryRequest.java

```diff
@@ -76,4 +81,10 @@ public String toString() {
         searchToString(b);
         return b.toString();
     }
+
+    @Override
+    public List<? extends IndicesRequest> subRequests() {
+        assert getSearchRequest() != null;
+        return Collections.singletonList(getSearchRequest());
```

Comment:
ok I remember now. The point of IndicesRequest and CompositeIndicesRequest is to return the indices that a request works against. when a request is composed of multiple operations, it should implement CompositeIndicesRequest. In this case delete by query reads from some indices as part of search, and writes as part of deletes. But what indices would it delete from? It is not possible to create a DeleteRequest that points to multiple indices, yet it is hard to predict all the deletions that will be performed as part of the request execution. I doubt that this request should implement CompositeIndicesRequest then.